src/Workspaces/Remote/ServiceHub/Services/FindUsages/RemoteFindUsagesService.cs

```diff
@@ -127,11 +128,18 @@ private int GetOrAddDefinitionItemId(DefinitionItem item)
                 }
             }
 
-            public ValueTask OnReferenceFoundAsync(SourceReferenceItem reference, CancellationToken cancellationToken)
+            public ValueTask OnReferencesFoundAsync(ImmutableArray<SourceReferenceItem> references, CancellationToken cancellationToken)
             {
-                var definitionItem = GetOrAddDefinitionItemId(reference.Definition);
-                var dehydratedReference = SerializableSourceReferenceItem.Dehydrate(definitionItem, reference);
-                return _callback.InvokeAsync((callback, cancellationToken) => callback.OnReferenceFoundAsync(_callbackId, dehydratedReference, cancellationToken), cancellationToken);
+                var dehydrated = new FixedSizeArrayBuilder<SerializableSourceReferenceItem>(references.Length);
+                foreach (var reference in references)
+                {
+                    var dehydratedReference = SerializableSourceReferenceItem.Dehydrate(
+                        GetOrAddDefinitionItemId(reference.Definition), reference);
```

Comment:
>GetOrAddDefinitionItemId [](http://example.com/codeflow?start=24&length=24)

I assume there wouldn't be an advantage to only locking _definitionItemToId once? (eg, do a first pass with a lock surrounding it that populates an fixedsizearraybuilder<int>)

Reply:
possibly.  but i'mnot seeing issues so far. so keeping things simple :)